sympy/functions/special/delta_functions.py

```diff
@@ -99,16 +101,17 @@ def simplify(self, x):
         """
         from sympy.polys.polyroots import roots
 
-        if not self.args[0].has(x) or (len(self.args) > 1 and self.args[1] != 0 ):
+        wrt = hints.pop('wrt')
```

Comment:
Thanks @asmeurer . Deleting pyc file rectified the error. Now it is coming this 
```py
In [5]: DiracDelta(x**2 + x - 2).expand(DiracDelta=True)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-ebc5bac3f771> in <module>()
----> 1 DiracDelta(x**2 + x - 2).expand(DiracDelta=True)

/home/ahappyidiot/Desktop/Gsoc/sympy/sympy/core/expr.pyc in expand(self, deep, modulus, power_base, power_exp, mul, log, multinomial, basic, **hints)
   2962             if use_hint:
   2963                 hint = '_eval_expand_' + hint
-> 2964                 expr, hit = Expr._expand_hint(expr, hint, deep=deep, **hints)
   2965 
   2966         while True:

/home/ahappyidiot/Desktop/Gsoc/sympy/sympy/core/expr.pyc in _expand_hint(expr, hint, deep, **hints)
   2903 
   2904         if hasattr(expr, hint):
-> 2905             newexpr = getattr(expr, hint)(**hints)
   2906             if newexpr != expr:
   2907                 return (newexpr, True)

/home/ahappyidiot/Desktop/Gsoc/sympy/sympy/functions/special/delta_functions.py in _eval_expand_DiracDelta(self, **hints)
    103 
    104         if 'wrt' not in hints:
--> 105           raise TypeError("The 'wrt' keyword is required for the DiracDelta hint")
    106 
    107         wrt = hints.pop('wrt')

TypeError: The 'wrt' keyword is required for the DiracDelta hint
```

But I have a doubt . Why didn't this work?
```py
try:
     wrt = hints.pop('wrt')
except KeyError:
     print ("KeyError: The \'wrt\' keyword is required for the DiracDelta hint")
```
Can you please explain a bit?